hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFs.java

```
@@ -1477,5 +1478,18 @@ public void setStoragePolicy(Path path, String policyName)
         throws IOException {
       throw readOnlyMountTable("setStoragePolicy", path);
     }
+
+    @Override
+    public Path getEnclosingRoot(Path path) throws IOException {
+      InodeTree.ResolveResult<AbstractFileSystem> res;
+      try {
+        res = fsState.resolve((path.toString()), true);
+      } catch (FileNotFoundException ex) {
+        throw new NotInMountpointException(path, "getEnclosingRoot");
```

You should propagate the original error as the cause of the thrown error.

---

This is the common notation / formatting. I am going to leave it as is for consistency.

---

I'd like it included. just because the current code doesn't do deep reporting, doesn't mean new code should retain the bad habit.